src/common/tests_dmg_helpers.c

```diff
@@ -583,13 +583,8 @@ parse_device_info(struct json_object *smd_dev, device_list *devices,
 	for (i = 0; i < dev_length; i++) {
 		dev = json_object_array_get_idx(smd_dev, i);
 
-		if (strlen(host + 2) <= DSS_HOSTNAME_MAX_LEN)
-			strcpy(devices[*disks].host, strtok(host, ":") + 1);
-		else {
-			D_ERROR("Hostname is larger than %d\n",
-				DSS_HOSTNAME_MAX_LEN);
-			return -DER_INVAL;
-		}
+		strncpy(devices[*disks].host, strtok(host, ":") + 1,
+			sizeof(devices[*disks].host) - 1);
```

Comment:
sizeof(devices[*disks].host) - 1) : Not clear why you are subtracting 1
strtok(host, ":") + 1, : Are you null terminating here... 
Not clear why 1 byte is added in strtok and one byte removed from sizeof... Maybe a comment about what is going on here could be helpful.

Reply:
I can remove +1 from strtok. I used -1 to avoid the buffer overwrite.

Reply:
Won't Coverity still complain about this?

jvolivie-desk1:[~/daos]$ cat foo.c
#include <stdio.h>
#include <string.h>
int main()
{
	char x[] = "";
	char y[] = ":";
	char z[] = ":abc";
	char m[] = "abc:";
	char n[] = "abc:cba";

	printf("%s\n", strtok(n, ":") + 1);
	fflush(stdout);
	printf("%s\n", strtok(m, ":") + 1);
	fflush(stdout);
	printf("%s\n", strtok(z, ":") + 1);
	fflush(stdout);
	printf("%s\n", strtok(y, ":") + 1);
	fflush(stdout);
	printf("%s\n", strtok(x, ":") + 1);
	fflush(stdout);
	return 0;
}

I get 
bc
bc
bc
Segmentation fault (core dumped)

Additionally, with strncpy, the string will not be null terminated if the length given would result in truncation.  Typically, it's good practice to explicitly make sure there is a 0 in the last character.

Reply:
I am not if that will complain about strtok or not. Let me find another way of copying it with null termination.